File: EDU-CF-R102/A.py
t = int(input())

while(t):

    n,d = list(map(int, input().split()))
    a = list(map(int, input().split()))
    a.sort()

    if a[-1] <= d:
        print('YES')
    elif a[0]+a[1] > d:
        print('NO')
    else:
        print('YES')

    t -= 1

File: EDU-CF-R102/A.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
  int tc;
  cin >> tc;
  while (tc--) {
    int n, d;
    cin >> n >> d;
    vector<int> a(n);
    for (int& x : a) cin >> x;
    sort(a.begin(), a.end());
    cout << (a.back() <= d || a[0] + a[1] <= d ? "YES" : "NO") << endl;
  }
}